### DISCUSSION TITLE
Will ref work in isolates in future versions?

### DISCUSSION THREAD

**Discussion by agufagit** [rrousselGit/riverpod#2889] [Q&A] [answered]
I need to run subscription in a long-lived isolate, I know that `ref` doesn't work in isolate in current version.  Just wondering when it will work or will it ever work?  If it's gonna be a long time or it will never work, I'm going to convert those providers to static functions.

**Comment by AhmedLSayed9** [comment, marked as answer]
You can access providers by defining a new ProviderContainer inside your isolate same as tests. That's a benefit of riverpod being able to work with only dart code.

**Reply by agufagit** [reply]
is there an example code I can look at?

Thank you

**Reply by AhmedLSayed9** [reply]
Check the "Dart only" code at https://docs-v2.riverpod.dev/docs/cookbooks/testing
